rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -86,6 +86,17 @@ pub struct HashJoinExec {
     build_side: Arc<Mutex<Option<JoinLeftData>>>,
     /// Shares the `RandomState` for the hashing algorithm
     random_state: RandomState,
+    /// Partitioning mode to use
+    mode: PartitionMode,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+/// Partitioning mode to use for hash join
+pub enum PartitionMode {
+    /// Left/right children are partitioned using the left and right keys
+    Partitioned,
+    /// Left side will be merged and collected into one partition
+    MergeLeft,
```

Comment:
maybe `CollectLeft` ? Every time I hear "Merge" I always think of "merging sorted streams" (e.g. the DataFusion `MergeExec` operator still confused me each time I see its name)

Reply:
Done